### PR TITLE
Extract scrape steam

### PR DESCRIPTION
**Pull request by Leon-JS02**
Changes made:
- Added a lambda handler script
- Added an optional time parameter that only scrapes titles from a particular date
- Restructured the listing parsing function
- Changed main parsing function of steam scraper to return json string instead of dict
- Added a main function to lambda handler for local Dockerless testing.

### CODE REVIEW

File: web_scraping/steam_scraper/scrape_steam.py [hunk 1]
```diff
@@ -27,12 +29,23 @@ def load_page_source(url: str) -> str:
 
 
 def get_page_listings(page_source: str) -> list[bs4.Tag]:
-    """Returns a list of Steam game listing objects."""
+    """Returns a list of Steam game listing objects released on a certain date."""
```
Comment (mehrdadhalali) [inline web_scraping/steam_scraper/scrape_steam.py:32]: Nice! Now we can scrape multiple days.

File: web_scraping/steam_scraper/lambda_handler.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        'statusCode': 200,
+        'body': {
+            'data': listings
+        }
+    }
+
+
+if __name__ == "__main__":
+    lambda_event = {
+        'time': "09-10-2024"
```
Comment (mehrdadhalali) [inline web_scraping/steam_scraper/lambda_handler.py:30]: Is this the parameter that changes?

**Review by mehrdadhalali — APPROVED** [review]

File: web_scraping/steam_scraper/lambda_handler.py (hunk 2 above)
Comment (Leon-JS02) [reply web_scraping/steam_scraper/lambda_handler.py:30]: The time parameter can be passed into the lambda to only gather games from that day. 
It defaults to None, which means all games are scraped.

Comment (mehrdadhalali) [reply web_scraping/steam_scraper/lambda_handler.py:30]: Could it be defaulted to today instead?

Comment (Leon-JS02) [reply web_scraping/steam_scraper/lambda_handler.py:30]: Done!

**Review by paulauus — APPROVED** [review]
All looks good, clear functions and import to handler. Looking forward to dataaaaa!

File: web_scraping/steam_scraper/lambda_handler.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+from scrape_steam import collect_and_parse_games
+
+# pylint: disable=W0613
+
+
+def lambda_handler(event, context):
+    """Triggered with the lambda. Optional time argument can be
+    passed in event: 'time'."""
+    time_arg = event.get('time', None)
+    try:
```
Comment (MichaelHassan) [inline web_scraping/steam_scraper/lambda_handler.py:14]: like this. Accounting for time of scrape

**Review by MichaelHassan — APPROVED** [review]
Happy with the changes made.

**Pull request merged**